File: 2 - Regression/2a - Simple Linear Regression/simple_linear_regression.R
# Simple Linear Regression

#Import the dataset
dataset = read.csv('Salary_Data.csv')

#Splitting the dataset into training and test set
library(caTools)
set.seed(123)
split = sample.split(dataset$Salary,SplitRatio = 2/3)
training_set = subset(dataset,split == TRUE)
test_set = subset(dataset,split == FALSE)

# Fitting simple linear regression to the Training set
regressor = lm(formula = Salary ~ YearsExperience,
               data = training_set)

# Predicting Test set results
y_pred = predict(regressor, newdata = test_set)

# Visualizing prediction of training set
# install.packages('ggplot2')
library(ggplot2)
ggplot() + 
    geom_point(aes(x = training_set$YearsExperience, y = training_set$Salary), 
               color = 'red') +
    geom_line(aes(x = training_set$YearsExperience, y = predict(regressor, newdata = training_set)),
              color = 'blue') +
    ggtitle('Salary vs Experience(Training Set)') +
    xlab('Years of Experience') +
    ylab('Salary')

# Visualizing prediction of test set
ggplot() + 
    geom_point(aes(x = test_set$YearsExperience, y = test_set$Salary), 
               color = 'red') +
    geom_line(aes(x = training_set$YearsExperience, y = predict(regressor, newdata = training_set)),
              color = 'blue') +
    ggtitle('Salary vs Experience(Test Set)') +
    xlab('Years of Experience') +
    ylab('Salary')

File: 2 - Regression/2a - Simple Linear Regression/simple_linear_regression.py
#	Simple Linear Regression

# Importing the libraries
import numpy as np 
import matplotlib.pyplot as plt
import pandas as pd 

# Import the dataset
dataset = pd.read_csv('Salary_Data.csv')
X = dataset.iloc[: , :-1].values
y = dataset.iloc[: , 1].values

# Splitting the dataset into training and test set
from sklearn.model_selection import train_test_split
X_train, X_test, y_train, y_test = train_test_split(X,y, test_size = 1/3, random_state = 0)

# Fitting Simple Linear Regression to the training set
from sklearn.linear_model import LinearRegression
regressor = LinearRegression()
regressor.fit(X_train, y_train)

# Predicting Test set result￼s
y_pred = regressor.predict(X_test)

# Visualizing Training set results
plt.scatter(X_train, y_train, color = "red")
plt.plot(X_train, regressor.predict(X_train), color = "blue")
plt.title("Salary vs Experience(Training Set)")
plt.xlabel("Years of Experience")
plt.ylabel("Salary")
plt.show()

# Visualizing Test set results
plt.scatter(X_test, y_test, color = "red")
plt.plot(X_train, regressor.predict(X_train), color = "blue")
plt.title("Salary vs Experience(Test Set)")
plt.xlabel("Years of Experience")
plt.ylabel("Salary")
plt.show()